devtools/create_achievements/steam_achievements.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+
+# This script takes two parameters
+#   - the first is the Steam game id, 
+#   - the second (optional) is the GOG Galaxy game id.
+#
+# Example:
+#   ./steam_achievements.py --steamid 631570 [--galaxyid 1845001352]
+#
+# It will extract the Steam stats information to create the ScummVM ini-files for achievements.dat file
+# For GOG versions of the games we assume that they are using same achievements as Steam versions
+
+import os
+import sys
+import codecs
+import argparse
+import requests
+from requests_html import HTMLSession
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--steamid", required=True, default="631570", type=int, help="The Steam game id")
+parser.add_argument("--saveasgalaxyid", type=int, help="The GOG Galaxy game id")
+parser.add_argument("-v", "--verbose", action="store_true")
+args = parser.parse_args()
+
+statsurl = "https://steamdb.info/app/{0}/stats/".format(args.steamid)
+if args.verbose:
+	sys.stderr.write("query {0}\n".format(statsurl))
+
+try:
+	session = HTMLSession()
+	response = session.get(statsurl)
+	game = response.html.xpath("//h1[@itemprop='name']/text()")
+
+	achievements_rows = response.html.xpath("//tr[starts-with(@id, 'achievement-')]/td")
+	achievements_columns = 3 # id, text, img
+	achievements_entries = int(len(achievements_rows) / achievements_columns)
+	if achievements_entries == 0:
+		sys.exit(127)
+
+	if args.verbose:
+		sys.stderr.write("found {0} achievements\n".format(achievements_entries))
+
+	stats_rows = response.html.xpath("//tr[starts-with(@id, 'stat-')]/td")
```

Comment:
As this is now more a less a tool that is needed in order to generate the dat file. We should add some more error checking and error messages here. If for example (as it happened in the past) steam changes the internal representation of their steamdb generated html - we might produce invalid files here. So it would be nice to add an option to error out if e.g. no achievement was found (as we expect to find them for the specified game ids). And also check that the format we expect is still found in the downloaded document. Please add exit codes != 0 for those error cases and make sure to check the return value in the `create_achievements.sh` script

Reply:
Fixed at b6049e16f2d855a589a178d84efaac64e25ef37f
Shell script now work with "set -e" and fails on first error.
Some error-checking added to refactored python script.